### PR TITLE
Improve robustness of the SDA function and add new features.

### PR DESCRIPTION
**Pull request by DongchenZ**
## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [x] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of CITATION.cff
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/assim.sequential/R/sda.enkf_MultiSite.R [hunk 1]
```diff
@@ -44,7 +44,9 @@ sda.enkf.multisite <- function(settings,
                                             debug = FALSE,
                                             pause = FALSE,
                                             Profiling = FALSE,
-                                            OutlierDetection=FALSE),
+                                            OutlierDetection=FALSE,
+                                            free_run = FALSE,
```
Comment (mdietze) [inline modules/assim.sequential/R/sda.enkf_MultiSite.R:48]: If you're adding new variables, make sure to update both the ROxygen and the changelog

File: modules/assim.sequential/R/sda.enkf_MultiSite.R [hunk 2]
```diff
@@ -425,21 +427,30 @@ sda.enkf.multisite <- function(settings,
         writeLines(runs.tmp[runs.tmp != ''], file.path(rundir, 'runs.txt'))
         paste(file.path(rundir, 'runs.txt'))  ## testing
         Sys.sleep(0.01)                       ## testing
-        PEcAn.workflow::start_model_runs(settings, write=settings$database$bety$write)
```
Comment (mdietze) [inline modules/assim.sequential/R/sda.enkf_MultiSite.R:428]: This should probably be an `if` to choose between versions, not a replacement.

File: modules/assim.sequential/R/sda.enkf_MultiSite.R [hunk 3]
```diff
@@ -425,21 +427,30 @@ sda.enkf.multisite <- function(settings,
...
-                         nens = nens, 
-                         read_restart_times = read_restart_times, 
-                         outdir = outdir, 
-                         t = t, 
-                         var.names = var.names, 
-                         my.read_restart = my.read_restart,
-                         restart_flag = restart_flag)
+        max_t <- 0
+        while (is.character(try(reads <- PEcAnAssimSequential:::build_X(out.configs = out.configs, 
+                                                                        settings = settings,
```
Comment (mdietze) [inline modules/assim.sequential/R/sda.enkf_MultiSite.R:437]: excess whitespace makes the code hard to read. Generally you want to align arguments vertically and keep lines to <80 characters wide (including whitespace)

File: modules/assim.sequential/R/sda.enkf_MultiSite.R (hunk 1 above)
Comment (mdietze) [reply modules/assim.sequential/R/sda.enkf_MultiSite.R:48]: Also, it doesn't look like either of these variables are actually used. Instead they are being accessed from within the `control` object. But they still need to be documented

**Review by mdietze — CHANGES_REQUESTED** [review]

File: modules/assim.sequential/R/sda.enkf_MultiSite.R (hunk 3 above)
Comment (DongchenZ) [reply modules/assim.sequential/R/sda.enkf_MultiSite.R]: Fixed.

File: modules/assim.sequential/R/sda.enkf_MultiSite.R (hunk 2 above)
Comment (DongchenZ) [reply modules/assim.sequential/R/sda.enkf_MultiSite.R:428]: Fixed.

**Review by mdietze — APPROVED** [review]

**Pull request merged**